--- a/kivy/kivy/__init__old.py	Mon Nov  2 23:04:12 2015
+++ b/kivy/kivy/__init__.py	Mon Nov  2 23:04:45 2015
@@ -249,7 +249,7 @@
 
 if not environ.get('KIVY_DOC_INCLUDE'):
     # Configuration management
-    user_home_dir = expanduser('~')
+    user_home_dir = expanduser('~').decode(sys.getfilesystemencoding())
     if platform == 'android':
         user_home_dir = environ['ANDROID_APP_PATH']
     elif platform == 'ios':
