--- filechooser-old.py	Tue Jun 28 14:26:06 2016
+++ filechooser.py	Tue Jun 28 14:47:18 2016
@@ -601,9 +601,8 @@
                 self.selection.append(entry.path)
         else:
             if _dir and not self.dirselect:
-                self.open_entry
                 return
-            self.selection = [entry.path, ]
+            self.selection = [abspath(join(self.path, entry.path)), ]
 
     def entry_released(self, entry, touch):
         '''(internal) This method must be called by the template when an entry
@@ -620,7 +619,7 @@
                 self.open_entry(entry)
             elif touch.is_double_tap:
                 if self.dirselect and self.file_system.is_dir(entry.path):
-                    self.open_entry(entry)
+                    return
                 else:
                     self.dispatch('on_submit', self.selection, touch)
 
@@ -637,7 +636,7 @@
             # If entry.path is to jump to previous directory, update path with
             # parent directory
             self.path = abspath(join(self.path, entry.path))
-            self.selection = []
+            self.selection = [self.path, ] if self.dirselect else []
 
     def _apply_filters(self, files):
         if not self.filters:
